lib/network/modules/components/Node.js

```diff
@@ -258,8 +267,20 @@ class Node {
     util.selectiveNotDeepExtend(fields, parentOptions, newOptions, allowDeletion);
 
     Node.checkMass(newOptions);
-    if (globalOptions.opacity !== undefined && newOptions.opacity === undefined) {
-      newOptions.opacity = globalOptions.opacity;
+
+    
+    if (parentOptions.opacity !== undefined) {
+      if (Number.isNaN(parentOptions.opacity) || !Number.isFinite(parentOptions.opacity) || parentOptions.opacity < 0 || parentOptions.opacity > 1) {
+        console.error("Invalid option for node opacity. Value must be between 0 and 1, found: " + parentOptions.opacity);
+        parentOptions.opacity = undefined;
+      }
+    }
+
+    if (newOptions.opacity !== undefined) {
+      if (Number.isNaN(newOptions.opacity) || !Number.isFinite(newOptions.opacity) || newOptions.opacity < 0 || newOptions.opacity > 1) {
+        console.error("Invalid option for node opacity. Value must be between 0 and 1, found: " + newOptions.opacity);
+        newOptions.opacity = undefined;
+      }
```

Comment:
It would be great to have this in a function rather than copy pasting it all over the place.